File: app/src/main/java/de/gymnasium_lappersdorf/gymlapapp/JobService.java
package de.gymnasium_lappersdorf.gymlapapp;

import android.app.job.JobInfo;
import android.app.job.JobParameters;
import android.app.job.JobScheduler;
import android.content.ComponentName;
import android.content.Context;
import android.os.PersistableBundle;

/**
 * Created by jonas on 12/3/2017.
 */

public class JobService extends android.app.job.JobService {


    //region constants

    private static final String EXTRA_TEXT = "extra_text";
    //endregion


    @Override
    public boolean onStartJob(JobParameters params) {

        String message = params.getExtras().getString(EXTRA_TEXT);
        //todo create Notification


        return false;

        
    }

    @Override
    public boolean onStopJob(JobParameters params) {
        return false;
    }

    //region helper methods

    //creates a new job that will execute at the given timestamp(in milli seconds) and show the given notification
    public static void createScedule(Context context, long timestamp, String notification){
        ComponentName componentName = new ComponentName(context, JobService.class);
        JobInfo.Builder builder = new JobInfo.Builder(0, componentName);
        builder.setMinimumLatency(timestamp-System.currentTimeMillis());
        PersistableBundle extra = new PersistableBundle();
        extra.putString(EXTRA_TEXT, notification);
        builder.setExtras(extra);
        JobScheduler jobScheduler =(JobScheduler) context.getSystemService(Context.JOB_SCHEDULER_SERVICE);
        if (jobScheduler != null) {
            jobScheduler.schedule(builder.build());
        }
    }



    //endregion
}

File: app/src/main/java/de/gymnasium_lappersdorf/gymlapapp/HausaufgabenPlaner/JobService.kt
package de.gymnasium_lappersdorf.gymlapapp.HausaufgabenPlaner

import android.app.job.JobInfo
import android.app.job.JobParameters
import android.app.job.JobScheduler
import android.content.ComponentName
import android.content.Context
import android.os.PersistableBundle

import de.gymnasium_lappersdorf.gymlapapp.MainActivity
import de.gymnasium_lappersdorf.gymlapapp.R

/**
 * Created by jonas on 12/3/2017.
 */

class JobService : android.app.job.JobService() {

    companion object {

        private const val EXTRA_TEXT = "extra_text"

        //creates a new job that will execute at the given timestamp(in milli seconds) and show the given notification
        fun createSchedule(context: Context, timestamp: Long, notification: String) : Int {
            val sharedPref = context.getSharedPreferences(context.getString(R.string.shared_pref_notification_id_key), Context.MODE_PRIVATE)
            var id = sharedPref.getInt(context.getString(R.string.shared_pref_notification_id), 0)
            id++
            val componentName = ComponentName(context, JobService::class.java)
            val builder = JobInfo.Builder(id, componentName)
            builder.setMinimumLatency(timestamp - System.currentTimeMillis())
            val extra = PersistableBundle()
            extra.putString(EXTRA_TEXT, notification)
            builder.setExtras(extra)
            val jobScheduler = context.getSystemService(Context.JOB_SCHEDULER_SERVICE) as JobScheduler
            jobScheduler.schedule(builder.build())

            with (sharedPref.edit()) {
                putInt(context.getString(R.string.shared_pref_notification_id), id)
                apply()
            }

            return id
        }
    }

    override fun onStartJob(params: JobParameters): Boolean {
        val message = params.extras.getString(EXTRA_TEXT)

        GymlapNotification(baseContext, message!!, MainActivity::class.java, params.jobId).show()
        return false
    }

    override fun onStopJob(params: JobParameters): Boolean {
        return false
    }
}
